docs/source/operators/deploy-kubernetes.md

```diff
@@ -15,248 +17,495 @@ The following sample kernel specifications apply to Kubernetes deployments:
+    - name: "http"
+      port: 8888
+      targetPort: 8888
+    #  nodePort: 32652 # optional nodePort
+    # The  port on which Enterprise Gateway will receive kernel connection info responses.
+    - name: "http-response"
+      port: 8877
+      targetPort: 8877
+    #  nodePort: 30481 # optional nodePort
+
+# Enterprise Gateway image name and tag to use from private registry.
+image: private.io/elyra/enterprise-gateway:dev
+
+deployment:
+# Update CPU/Memory as needed
+  resources:
+    limits:
+      cpu: 2
+      memory: 10Gi
+    requests:
+      cpu: 1
+      memory: 2Gi
+  # Update to deploy multiple replicas of EG.
+  replicas: 2
+
+kip:
+  enabled: false # turn this off, if runnning DaemonSets is restricted by your cluster Administrator
+  # Kernel Image Puller image name and tag to use from private registry.
+  image: private.io/elyra/kernel-image-puller:dev
+
+ingress:
+  enabled: false
+```
 
-#### Create the Enterprise Gateway kubernetes service and deployment
+### Deploy with helm
 
-From the master node, create the service and deployment using the yaml file from a source release or the git repository:
+Add values file to your helm command and apply to the cluster replacing [PLACEHOLDER] with appropriate values for your environment:
 
 ```bash
-kubectl apply -f etc/kubernetes/enterprise-gateway.yaml
+helm  upgrade --install \
+  jupyter-e-gw \
+  jupyter-server/enterprise_gateway/etc/kubernetes/helm/enterprise-gateway \
+   --kube-context [mycluster-context-name] \
+   --namespace [namespace-name] \
+   -f values-full.yaml
+```
+
+if you are using private registry add setting base64 encoded secret value to you command:
+`--set imagePullSecretsCreate.secrets[0].data="UHJvZCBTZWNyZXQgSW5mb3JtYXRpb24K"`
+
+### Deploy with kubectl
+
+Choose this deployment option if you want to deploy directly from Kubernetes template files with kubectl, rather than using a package manager like helm.
 
-service "enterprise-gateway" created
-deployment "enterprise-gateway" created
+Add values file to your helm command and generate `yaml` files replacing [PLACEHOLDER] with appropriate values for your environment:
+
+```bash
+helm template \
+  --output-dir [/tmp/mydeployment] \
+  jupyter-e-gw \
+  etc/kubernetes/helm/enterprise-gateway \
+   --namespace [namespace-name] \
+   -f values-full.yaml
 ```
 
-#### Uninstalling Enterprise Gateway
+if you are using private registry add setting base64 encoded secret value to you command:
+`--set imagePullSecretsCreate.secrets[0].data="UHJvZCBTZWNyZXQgSW5mb3JtYXRpb24K"`
 
-To shut down Enterprise Gateway issue a delete command using the previously mentioned global label `app=enterprise-gateway`
+Now you can review generated `yaml` files and apply them to your Kubernetes cluster:
 
 ```bash
-kubectl delete all -l app=enterprise-gateway
+kubectl apply -f /tmp/mydeployment/enterprise-gateway/templates/
 ```
 
-or simply delete the namespace
+_**NOTE: Do not store secrets in the source control.**_
```

Comment:
I think this is great advice for an "important" callout...
```suggestion
```{important}
Never store secrets in your source code control files!
\`\`\`
```
Note that the trailing backtick-tuple is not meant to be escaped but they conflict with the closing backtick-tuple in the GH suggestion.  (Do you happen to know how to deal with that?)